nzedb/Forum.php

```diff
@@ -23,119 +25,175 @@ public function __construct(array $options = [])
 		$this->pdo = ($options['Settings'] instanceof DB ? $options['Settings'] : new DB());
 	}
 
+	/**
+	 * @param     $parentid
+	 * @param     $userid
+	 * @param     $subject
+	 * @param     $message
+	 * @param int $locked
+	 * @param int $sticky
+	 * @param int $replies
+	 *
+	 * @return false|int|string
+	 */
 	public function add($parentid, $userid, $subject, $message, $locked = 0, $sticky = 0, $replies = 0)
 	{
-		if ($message == "") {
+		if ($message === '') {
 			return -1;
 		}
 
-		if ($parentid != 0) {
+		if ($parentid !== 0) {
 			$par = $this->getParent($parentid);
-			if ($par == false) {
+			if ($par === false) {
 				return -1;
 			}
 
-			$this->pdo->queryExec(sprintf("UPDATE forum_posts SET replies = replies + 1, updateddate = NOW() WHERE id = %d", $parentid));
+			$this->pdo->queryExec(sprintf('UPDATE forum_posts SET replies = replies + 1, updateddate = NOW() WHERE id = %d', $parentid));
 		}
 
 		return $this->pdo->queryInsert(
-			sprintf("
+			sprintf('
 				INSERT INTO forum_posts (forumid, parentid, user_id, subject, message, locked, sticky, replies, createddate, updateddate)
-				VALUES (1, %d, %d, %s, %s, %d, %d, %d, NOW(), NOW())",
+				VALUES (1, %d, %d, %s, %s, %d, %d, %d, NOW(), NOW())',
 				$parentid, $userid, $this->pdo->escapeString($subject), $this->pdo->escapeString($message), $locked, $sticky, $replies
 			)
 		);
 	}
 
+	/**
+	 * @param $parent
+	 *
+	 * @return array|bool
+	 */
 	public function getParent($parent)
 	{
 		return $this->pdo->queryOneRow(
 			sprintf(
-				"SELECT forum_posts.*, users.username FROM forum_posts LEFT OUTER JOIN users ON users.id = forum_posts.user_id WHERE forum_posts.id = %d",
+				'SELECT forum_posts.*, users.username FROM forum_posts LEFT OUTER JOIN users ON users.id = forum_posts.user_id WHERE forum_posts.id = %d',
 				$parent
 			)
 		);
 	}
 
+	/**
+	 * @param $parent
+	 *
+	 * @return array
+	 */
 	public function getPosts($parent)
 	{
 		return $this->pdo->query(
-			sprintf("
+			sprintf('
 				SELECT forum_posts.*, users.username
 				FROM forum_posts
 				LEFT OUTER JOIN users ON users.id = forum_posts.user_id
 				WHERE forum_posts.id = %d OR parentid = %d
 				ORDER BY createddate ASC
-				LIMIT 250",
+				LIMIT 250',
 				$parent,
 				$parent
 			)
 		);
 	}
 
+	/**
+	 * @param $id
+	 *
+	 * @return array|bool
+	 */
 	public function getPost($id)
 	{
-		return $this->pdo->queryOneRow(sprintf("SELECT * FROM forum_posts WHERE id = %d", $id));
+		return $this->pdo->queryOneRow(sprintf('SELECT * FROM forum_posts WHERE id = %d', $id));
 	}
 
+	/**
+	 * @return int
+	 */
 	public function getBrowseCount()
 	{
-		$res = $this->pdo->queryOneRow(sprintf("SELECT COUNT(id) AS num FROM forum_posts WHERE parentid = 0"));
-		return ($res === false ? 0 : $res["num"]);
+		$res = $this->pdo->queryOneRow(sprintf('SELECT COUNT(id) AS num FROM forum_posts WHERE parentid = 0'));
+		return ($res === false ? 0 : $res['num']);
 	}
 
+	/**
+	 * @param $start
+	 * @param $num
```

Comment:
We should know the type of both of these. As they're going to the LIMIT clause they should be integers. We should at least type them here and definitely consider forcing the check in the signature